src/extension.ts

```diff
@@ -63,3 +86,76 @@ const getComponentName = (): Thenable<string | undefined> =>
         : "Please enter a valid component name";
     },
   });
+
+/**
+ * Check availability of Angular CLI on the command line
+ * @param directory Directory which will use the Angular CLI
+ */
+const checkAngularCli = (directory: string) => {
+  try {
+    execSync("ng --version", { cwd: directory });
+  } catch (error) {
+    throw Error("Could not find Angular CLI, Error: " + error);
+  }
+};
+
+/**
+ * Get the name of the directory of the current file
+ * @returns Name of the directory
+ */
+const getDirectoryName = () => {
+  const fileName = vscode.window.activeTextEditor?.document.fileName;
+  if (!fileName) {
+    throw Error("Filename not known!");
+  }
+  return path.dirname(fileName);
+};
+
+/**
+ * Callback that returns the actual progress of VS Code to generate a component
+ * @param componentDirectory Directory of the current component
+ * @param componentName Name of the component which will be generated
+ * @param useNpx If Angular CLI is not installed on the machine
+ * @returns Promise that resolves if the component has beed created successfully
+ */
+const generateComponentProgress = (
+  componentDirectory: string,
+  componentName: string,
+  useNpx: boolean
+) => async (
+  progress: vscode.Progress<{
+    message?: string | undefined;
+    increment?: number | undefined;
+  }>
+): Promise<void> => {
+  progress.report({ increment: 0 });
+
+  /** Angular CLI command */
+  const command =
+    (useNpx ? "npx " : "") + `ng generate component ${componentName}`;
+
+  exec(
+    command,
+    { cwd: componentDirectory },
+    async (err, stdout: string, stderr: string) => {
+      await Promise.resolve();
+
+      progress.report({ increment: 100 });
+      vscode.window.showInformationMessage(
+        `Component ${componentName} generated successfully`
```

Comment:
This message is shown regardless if it was successful or not. Please only show the message if the component creation was successful.

Reply:
Moved it to the else block below, you're right!